internal/daysteps/daysteps.go

```diff
@@ -1,13 +1,67 @@
 package daysteps
 
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/Yandex-Practicum/tracker/internal/personaldata"
+	"github.com/Yandex-Practicum/tracker/internal/spentenergy"
+)
+
 type DaySteps struct {
-	// TODO: добавить поля
+	Steps    int
+	Duration time.Duration
+	personaldata.Personal
 }
 
+var (
+	ErrWrongInfo     = errors.New("incorrect data")
+	ErrStepsLessZero = errors.New("steps cannot be less than zero")
+	ErrDurLessZero   = errors.New("duration cannot be less than zero")
+)
+
+// Метод парсит строку с данными формата "678,0h50m" и записывает данные в соответствующие поля структуры DaySteps.
 func (ds *DaySteps) Parse(datastring string) (err error) {
-	// TODO: реализовать функцию
+	dataSlice := strings.Split(datastring, ",")
+
+	if len(dataSlice) != 2 || datastring == "" {
+		return ErrWrongInfo
+	}
+
+	ds.Steps, err = strconv.Atoi(dataSlice[0])
+	if err != nil {
+		return err
+	}
+	if ds.Steps <= 0 {
+		return ErrStepsLessZero
+	}
+
+	ds.Duration, err = time.ParseDuration(dataSlice[1])
+	if err != nil {
+		return err
```

Comment:
Лучше оборачивать ошибки
```suggestion
		return fmt.Errorf("invalid duration: %w", err)
```